### PR TITLE
support create/drop unique index

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6846 

## What this PR does / why we need it:

1.Because the code of index cn is not completed before, some cases in bvt are not exposed, so mark here
2.When creating an index table, the data of the original table is written to the index table through the reader interface. Later, it will be changed to insert... select

### CODE REVIEW

File: pkg/catalog/catalog.go [hunk 1]
```diff
@@ -188,6 +190,18 @@ func genCreateTables(rows [][]any) []CreateTable {
...
 }
 
+func genUpdateConstraint(rows [][]any) []UpdateConstraint {
+	cmds := make([]UpdateConstraint, len(rows))
+	for i, row := range rows {
+		cmds[i].TableId = row[MO_TABLES_REL_ID_IDX].(uint64)
+		cmds[i].DatabaseId = row[MO_TABLES_RELDATABASE_ID_IDX].(uint64)
+		cmds[i].TableName = string(row[MO_TABLES_REL_NAME_IDX].([]byte))
+		cmds[i].DatabaseName = string(row[MO_TABLES_RELDATABASE_IDX].([]byte))
+		cmds[i].Constraint = row[4].([]byte)
```
Comment (fengttt) [inline pkg/catalog/catalog.go:200]: 4?  should this be a defined const as well?

**Review by fengttt — COMMENTED** [review]
when user create index, it will build the index table, correct?   in a transaction?   what if there is a conflict?

**Review by aressu1985 — APPROVED** [review]

**Review by jiangxinmeng1 — APPROVED** [review]

**Comment by iamlinjunhong** [comment]
> when user create index, it will build the index table, correct? in a transaction? what if there is a conflict?

it will build the index table. And will rollback if conflict. Table creation and dml are same transaction.

**Review by aptend — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Review by aunjgr — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**